crates/wasmi/src/table/mod.rs

```diff
@@ -535,12 +586,13 @@ impl Table {
         mut ctx: impl AsContextMut,
         delta: u32,
         init: Value,
+        limiter: &mut ResourceLimiterRef<'_>,
     ) -> Result<u32, TableError> {
         ctx.as_context_mut()
             .store
             .inner
             .resolve_table_mut(self)
-            .grow(delta, init)
+            .grow(delta, init, limiter)
```

Comment:
Same as with `Memory::grow`: We shouldn't alter the API of user facing types.
Instead (with the addition of the aforementioned getters) we can do the following:
```rust
pub fn grow(
    &self,
    mut ctx: impl AsContextMut,
    delta: u32,
    init: Value,
) -> Result<u32, TableError> {
    let (store_inner, mut limiter) = ctx
        .as_context_mut()
        .store
        .store_inner_and_resource_limiter_ref();
    store_inner
        .resolve_table_mut(self)
        .grow(delta, init, &mut limiter)
}
```

Reply:
Ok (changed in my local branch).

I should mention that both `MemoryEntity::new` and `TableEntity::new` methods do change still. Not sure if those concern you (since they're not `wasmtime` types). They are public.

Reply:
Both `MemoryEntity` and `TableEntity` are crate-private definitions therefore changing their APIs is not a concern to me.